### PR TITLE
execcmd manager init task: Windows functionality added

### PR DESCRIPTION
**Pull request by arun-annamalai**
### Summary

This pull request implements the ECS agent init_task functionality so that the SSM agent has the necessary bind mounts and binaries to run. 

### Implementation details

The common variables are pulled into the manager_init_task.go file and then the Linux code is kept in its file. The windows file is where the bulk of the new implementation is. Mainly, Windows stores its configs in a different way since it cannot bind mount files (This is a [limitation](https://github.com/moby/moby/issues/30555) of docker windows). In addition, we need to bind mount the SSM plugins since the windows agent needs those. Because the dirs are bind mounted, they will be cached on a directory level (by concatenating the contents of the dir and hashing them), on Linux this is done in a file level. Lastly, when mounting in Windows host file paths/folders are not automatically created by Docker so they need to be manually created before mounting. 

### Testing

unit tests

New tests cover the changes: yes 
integ tests will be in following PR
### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: agent/engine/execcmd/manager_init_task_windows.go [hunk 1]
```diff
@@ -16,18 +16,209 @@
 package execcmd
 
 import (
```
Comment (ubhattacharjya) [inline agent/engine/execcmd/manager_init_task_windows.go:18]: Should unit tests and integ tests be included for these changes?

Comment (arun-annamalai) [reply agent/engine/execcmd/manager_init_task_windows.go:18]: yes, I will add them in now

File: agent/engine/execcmd/manager_init_task_windows.go [hunk 2]
```diff
@@ -16,18 +16,214 @@
...
+    </outputs>
+    <formats>
+        <format id="fmterror" format="%Date %Time %LEVEL [%FuncShort @ %File.%Line] %Msg%n"/>
+        <format id="fmtdebug" format="%Date %Time %LEVEL [%FuncShort @ %File.%Line] %Msg%n"/>
+        <format id="fmtinfo" format="%Date %Time %LEVEL %Msg%n"/>
+    </formats>
+</seelog>`
+)
+
+func (m *manager) InitializeContainer(taskId string, container *apicontainer.Container, hostConfig *dockercontainer.HostConfig) (rErr error) {
```
Comment (angelcar) [inline agent/engine/execcmd/manager_init_task_windows.go:99]: I'd prefer not to fork this whole method as it has many commonalities with linux. Instead we should probably just add a sub-method called `addRequiredBindMounts` or something similar. That way we can preserve a single method (i.e. reuse as much as possible) while still being able to configure windows in a windows-specific way

Comment (arun-annamalai) [reply agent/engine/execcmd/manager_init_task_windows.go:99]: will refactor and push

File: agent/engine/execcmd/manager_init_task_linux.go [hunk 3]
```diff
@@ -107,160 +75,89 @@ var (
...
-	limit, err := strconv.Atoi(limitStr)
-	if err != nil { // This means ACS send a limit that can't be converted to an int
-		return limit
-	}
-	if limit <= 0 {
-		limit = defaultSessionLimit
-	}
-	return limit
-}
+//func (m *manager) InitializeContainer(taskId string, container *apicontainer.Container, hostConfig *dockercontainer.HostConfig) (rErr error) {
```
Comment (angelcar) [inline agent/engine/execcmd/manager_init_task_linux.go:84]: are we ready to remove this?

File: agent/engine/execcmd/manager_init_task_windows.go [hunk 4]
```diff
@@ -16,18 +16,236 @@
...
+        <format id="fmtdebug" format="%Date %Time %LEVEL [%FuncShort @ %File.%Line] %Msg%n"/>
+        <format id="fmtinfo" format="%Date %Time %LEVEL %Msg%n"/>
+    </formats>
+</seelog>`
 )
 
-// InitializeContainer adds the necessary bind mounts in order for the ExecCommandAgent to run properly in the container
-// Note: exec cmd agent is a linux feature, thus implemented here as a no-op.
-func (m *manager) InitializeContainer(taskId string, container *apicontainer.Container, hostConfig *dockercontainer.HostConfig) error {
+//func (m *manager) InitializeContainer(taskId string, container *apicontainer.Container, hostConfig *dockercontainer.HostConfig) (rErr error) {
```
Comment (angelcar) [inline agent/engine/execcmd/manager_init_task_windows.go:83]: same as above, let's remove this if we don't need it anymore

File: agent/engine/execcmd/manager_init_task_linux.go (hunk 3 above)
Comment (arun-annamalai) [reply agent/engine/execcmd/manager_init_task_linux.go]: yup, removed

File: agent/engine/execcmd/manager_init_task_windows.go (hunk 4 above)
Comment (arun-annamalai) [reply agent/engine/execcmd/manager_init_task_windows.go]: yup, removed

**Review by angelcar — APPROVED** [review]

**Review by sharanyad — APPROVED** [review]

**Pull request merged**